### ISSUE TITLE
ReduceLROnPlateau potential issue

### ISSUE THREAD

**Issue by btclement** [PaulRitsche/DeepACSA#36]
First of all, thank you for your work.
I am new in the field so take my statement with caution. I may have found an issue with the learning rate decay.

In your paper you state : "The learning rate was reduced by a factor of 0.1 after 10 epochs of loss stagnation."

And in your code you define : https://github.com/PaulRitsche/DeepACSA/blob/4739fe6b80455fd8359d0522e1c30fb796f62ad8/Deep_ACSA/gui_helpers/model_training.py#L512-L514

But when defining ReduceLROnPlateau, you pass it as  :

> min_lr: Float. Lower bound on the learning rate

https://github.com/PaulRitsche/DeepACSA/blob/4739fe6b80455fd8359d0522e1c30fb796f62ad8/Deep_ACSA/gui_helpers/model_training.py#L624-L626

Which, for me, means it won't reduce at all since the lower bound is reached from the start

Moreover, the patience for Early stopping is set at 8 epochs, while ReduceLROnPlateau's patience is set at 10.
With the same min_delta, the training will stop before learning decay may activate, won't it ?

**Comment by PaulRitsche** [comment]
Dear @btclement, 

thanks for pointing this out. 
You are correct in you assumption that this is a bug. 
We are currently working a new version in which we will fix the issue. This will probably be done by September. 

If you cannot wait that long, I will push a path to the **towads_v0.3.2 branch** 
But I cannot guarantee full functionality there. 

BW, Paul

**Comment by btclement** [comment]
Dear @PaulRitsche,

Thank you for answering so fast. I'm happy I could help improve this model.

I may have spotted an other issue while I'm at it :
https://github.com/PaulRitsche/DeepACSA/blob/4739fe6b80455fd8359d0522e1c30fb796f62ad8/Deep_ACSA/gui_helpers/model_training.py#L114-L121

The 2nd conv is applied on **inputs** instead of **x** which would mean the first conv is ignored ?
I guess it's a copy/paste oversight.

BW, Clément

**Comment by PaulRitsche** [comment]
Dear @btclement, 

please excude my late response.
Again, you are correct in your assumption, thanks for being so attentive. 

If you want, you can file a pull-request with the fixes. 
Otherwise, they will be dealt with in the next release. 

BW, Paul
